src/workerd/api/global-scope.h

```diff
@@ -708,7 +713,13 @@ class ServiceWorkerGlobalScope: public WorkerGlobalScope {
       JSG_NESTED_TYPE(URL);
       JSG_NESTED_TYPE(URLSearchParams);
     }
-    JSG_NESTED_TYPE(URLPattern);
+
+    // We conditionally enable a more spec compliant URLPattern to avoid any breakages.
+    if (util::Autogate::isEnabled(util::AutogateKey::URLPATTERN)) {
+      JSG_NESTED_TYPE(urlpattern::URLPattern);
```

Comment:
You'll want to use the `JSG_NESTED_TYPE_NAMED` variant here, otherwise the `urlpattern:` bit will end up part of the name here.